closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -257,17 +266,24 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         }
     } else {
         openssl_version = (*OSSL_version)(0); /* get OPENSSL_VERSION */
-        /* Ensure the OpenSSL version is "OpenSSL 1.1.x". */
-        if (0 != strncmp(openssl_version, OPENSSL_VERSION_1_1, strlen(OPENSSL_VERSION_1_1))) {
-            /* fprintf(stderr, "Incompatable OpenSSL version: %s\n", openssl_version); */
-            /* fflush(stderr); */
+        /* Ensure the OpenSSL version is "OpenSSL 1.1.x" or "OpenSSL 3.0.x". */
+        if (0 != strncmp(openssl_version, OPENSSL_VERSION_1_1, strlen(OPENSSL_VERSION_1_1)) && 0 != strncmp(openssl_version, OPENSSL_VERSION_3_0, strlen(OPENSSL_VERSION_3_0))) {
```

Comment:
Please add parentheses around each argument to `&&` and split the line before `&&`.

Reply:
also change "OpenSSL 3.0.x" to "OpenSSL 3.x.x"

Reply:
Edited